src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(1L);
+        user.setLogin("rightLogin");
+        user.setPassword("rightPassword");
+        user.setAge(25);
+    }
+
+    @AfterEach
+    void tearDown() {
+        ((StorageDaoImpl) storageDao).clear();
+    }
+
+    @Test
+    void register_rightUser_ok() {
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+
+        User user2 = new User();
+        user2.setId(2L);
+        user2.setLogin("123456");
+        user2.setPassword("123456");
+        user2.setAge(18);
+        User actual2 = registrationService.register(user2);
+        assertEquals(user2, actual2);
+    }
+
+    @Test
+    void register_userWithExistLogin_notOk() {
+        Storage.people.add(user);
+        User userWithExistLogin = user;
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(userWithExistLogin);
+        });
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_userWithNullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userWithWrongLogin_notOk() {
+        user.setLogin("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setLogin("log");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setLogin("login");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userWithNullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userWithWrongPassword_notOk() {
+        user.setPassword("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setPassword("pas");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setPassword("passw");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
```

Comment:
You have tested for empty and short passwords, but you should also add a test for a password that is exactly at the minimum length requirement to ensure that edge cases are fully covered.